File: src/main/java/toast/pine/Adapters/SocialProfileAdapter.java
package toast.pine.Adapters;

import org.bukkit.persistence.PersistentDataAdapterContext;
import org.bukkit.persistence.PersistentDataType;
import org.jetbrains.annotations.NotNull;
import toast.pine.SocialSystem.PlayerSocial;

public class SocialProfileAdapter implements PersistentDataType<PlayerSocial, PlayerSocial> {
    @Override
    public @NotNull Class<PlayerSocial> getPrimitiveType() {
        return PlayerSocial.class;
    }

    @Override
    public @NotNull Class<PlayerSocial> getComplexType() {
        return PlayerSocial.class;
    }

    @Override
    public @NotNull PlayerSocial toPrimitive(@NotNull PlayerSocial complex, @NotNull PersistentDataAdapterContext context) {
        return complex;
    }

    @Override
    public @NotNull PlayerSocial fromPrimitive(@NotNull PlayerSocial primitive, @NotNull PersistentDataAdapterContext context) {
        return primitive;
    }
}

File: src/main/java/pine/toast/Adapters/SocialProfileAdapter.kt
package toast.pine.Adapters

import org.bukkit.persistence.PersistentDataAdapterContext
import org.bukkit.persistence.PersistentDataType
import toast.pine.socialSystem.PlayerSocial

class SocialProfileAdapter : PersistentDataType<PlayerSocial, PlayerSocial> {
    @get:JvmName("getPrimitiveTypePlayerSocial")
    private val primitiveType: Class<PlayerSocial>
        get() = PlayerSocial::class.java
    @get:JvmName("getComplexTypePlayerSocial")
    private val complexType: Class<PlayerSocial>
        get() = PlayerSocial::class.java

    override fun toPrimitive(complex: PlayerSocial, context: PersistentDataAdapterContext): PlayerSocial {
        return complex
    }

    override fun fromPrimitive(primitive: PlayerSocial, context: PersistentDataAdapterContext): PlayerSocial {
        return primitive
    }

    override fun getPrimitiveType(): Class<PlayerSocial> {
        return primitiveType
    }

    override fun getComplexType(): Class<PlayerSocial> {
        return complexType
    }
}
